src/main/java/com/example/WatchNext/controller/AuthController.java

```diff
@@ -0,0 +1,100 @@
+package com.example.WatchNext.controller;
+
+import com.example.WatchNext.model.Role;
+import com.example.WatchNext.model.Users;
+import com.example.WatchNext.payload.request.LoginRequest;
+import com.example.WatchNext.payload.request.SignupRequest;
+import com.example.WatchNext.payload.response.JwtResponse;
+import com.example.WatchNext.payload.response.MessageResponse;
+import com.example.WatchNext.repositories.RoleRepository;
+import com.example.WatchNext.repositories.UserRepository;
+import com.example.WatchNext.security.jwt.JwtUtils;
+import com.example.WatchNext.security.services.UserDetailsImpl;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.Collection;
+
+
+@CrossOrigin(origins = "*", maxAge = 3600)
+@RestController
+@RequestMapping("/api/auth")
+public class AuthController {
+    @Autowired
+    AuthenticationManager authenticationManager;
+
+    @Autowired
+    UserRepository userRepository;
+
+    @Autowired
+    RoleRepository roleRepository;
+
+    @Autowired
+    PasswordEncoder encoder;
+
+    @Autowired
+    JwtUtils jwtUtils;
+
+    @PostMapping("/signin")
+    public ResponseEntity<?> authenticateUser(@RequestBody LoginRequest loginRequest) {
+
+        Authentication authentication = authenticationManager.authenticate(
+                new UsernamePasswordAuthenticationToken(loginRequest.getUsername(), loginRequest.getPassword()));
+
+        SecurityContextHolder.getContext().setAuthentication(authentication);
+        String jwt = jwtUtils.generateJwtToken(authentication);
+
+        UserDetailsImpl userDetails = (UserDetailsImpl) authentication.getPrincipal();
+        Collection<? extends GrantedAuthority> roles = userDetails.getAuthorities();
+
+        return ResponseEntity.ok(new JwtResponse(jwt,
+                userDetails.getId(),
+                userDetails.getUsername(),
+                userDetails.getEmail(),
+                roles));
+    }
+
+    @PostMapping("/signup")
+    public ResponseEntity<?> registerUser(@RequestBody SignupRequest signUpRequest) {
```

Comment:
this method is badly placed, it should be under its own controller: UserController
authenticating a user has nothing to do with creating a new user